File: leetcode/c++/1480_running_sum_of_1d_array.cpp
// https://leetcode.cn/problems/running-sum-of-1d-array/description

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        if (nums.size() < 2)    
            return nums;
        vector<int>dp(nums.size());
        dp[0] = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            dp[i] = dp[i-1] + nums[i];
        }
        return dp;
    }
};

// 时间复杂度和空间复杂度为O(n)

File: leetcode/golang/1480_running_sum_of_1d_array.go
func runningSum(nums []int) []int {
    if len(nums) < 2 {
        return nums
    }
    var dp []int = make([]int, len(nums))
    dp[0] = nums[0]
    for i := 1; i < len(nums); i++ {
        dp[i] = dp[i-1] + nums[i]
    }
    return dp
}